app/customer.py

```diff
@@ -0,0 +1,71 @@
+import json
+
+import math
+
+from dataclasses import dataclass
+
+import app.shop as mall
+from app.cart import Cart
+from app.car import Car
+
+
+@dataclass
+class Customer:
+    name: str
+    product_cart: list[Cart]
+    location: list[int]
+    money: float
+    car: Car
+
+    def calculate_trip_to_shop(self, shop: mall) -> float:
+        fuel_consumption = self.car.fuel_consumption
+        distance = self._count_distance(shop.location)
+        trip_cost = round(fuel_consumption * distance * 2
+                          / 100 * self._get_fuel_price(), 2)
+
+        return trip_cost
+
+    def print_report(self, shops: list[mall]) -> str:
+        shop_dict = {}
+        result = f"{self.name} has {self.money} dollars\n"
+
+        for shop in shops:
+            trip_sum = (self.calculate_trip_to_shop(shop)
+                        + shop.print_receipt(self)[1])
```

Comment:
On every iteration, you call the method calculate_trip_to_shop which calls on every iteration the _get_fuel_price() method which opens and reads the same data from the file on every iteration as well. It's not good, just get this data with only one opening the file.

Reply:
fixed